File: solutions/349. Intersection of Two Arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        s=set()
        res=set()
        for i in nums1:
            if i not in s:
                s.add(i)
        for j in nums2:
            if j in s:
                res.add(j)
        return res
            
        

File: solutions/349. Intersection of Two Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
       
        Set<Integer> s1 = new HashSet<>();
        Set<Integer> s2 = new HashSet<>();
        for (Integer num:nums1) s1.add(num);
       
        for (Integer num: nums2)  {
            if (s1.contains(num)) {
                s2.add(num);
            }
        }
        
        
        int[] res = new int[s2.size()];
        int idx =0 ;
        for (int s: s2) {
            res[idx++]=s;
        }
        
        return res;
        
    }
}
